.github/workflows/cross-compile.yaml

```diff
@@ -0,0 +1,312 @@
+
+
+jobs:
+
+  cross-compile:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        # Use bash shell as default
+        shell: bash
+
+    strategy:
+      # Allow other runners in the matrix to continue if some fail
+      fail-fast: false
+
+      matrix:
+        arch: [armhf, ppc64el, s390x, riscv64]
+        include:
+          - arch: armhf
+            target-triple: arm-linux-gnueabihf
+            # cross-toolchain: g++-9-arm-linux-gnueabihf
+            # fortran-cross-toolchain: gfortran-14-arm-linux-gnueabihf
+            # cc: "arm-linux-gnueabihf-gcc-9"
+            # cxx: "arm-linux-gnueabihf-g++-9"
+            # fc: "arm-linux-gnueabihf-gfortran-14"
+            ccache-max: 64M
+          - arch: ppc64el
+            target-triple: powerpc64le-linux-gnu
+            # cross-toolchain: g++-9-powerpc64le-linux-gnu 
+            # fortran-cross-toolchain: gfortran-14-powerpc64le-linux-gnu
+            # cc: "powerpc64le-linux-gnu-gcc-9"
+            # cxx: "powerpc64le-linux-gnu-g++-9"
+            # fc: "powerpc64le-linux-gnu-gfortran-14"
+            ccache-max: 64M
+          - arch: s390x
+            target-triple: s390x-linux-gnu
+            # cross-toolchain: g++-9-s390x-linux-gnu 
+            # fortran-cross-toolchain: gfortran-14-s390x-linux-gnu
+            # cc: "s390x-linux-gnu-gcc-9"
+            # cxx: "s390x-linux-gnu-g++-9"
+            # fc: "s390x-linux-gnu-gfortran-14"
+            ccache-max: 64M
+          - arch: riscv64
+            target-triple: riscv64-linux-gnu
+            # cross-toolchain: g++-9-riscv64-linux-gnu 
+            # fortran-cross-toolchain: gfortran-14-riscv64-linux-gnu
+            # cc: "riscv64-linux-gnu-gcc-9"
+            # cxx: "riscv64-linux-gnu-g++-9"
+            # fc: "riscv64-linux-gnu-gfortran-14"
+            ccache-max: 64M
+
+    name: cross-compile (${{ matrix.arch }})
+
+    env:
+      CC: clang -target ${{ matrix.target-triple }}
+      CXX: clang++ -target ${{ matrix.target-triple }}
+      FC: ${{ matrix.target-triple }}-gfortran-14
+      LD_LIBRARY_PATH: "/usr/${{ matrix.target-triple }}/lib"
+      LIBRARY_PATH: "/usr/${{ matrix.target-triple }}/lib"
+
+    steps:
+      - name: get CPU information
+        run: lscpu
+
+      - name: install some aux utilities
+        run: |
+          echo "::group::Update package indexes"
+          sudo apt update
+          echo "::endgroup::"
+          echo "::group::Install cmake make ccache automake libtool qemu-user-binfmt"
+          sudo apt -y install \
+            cmake \
+            make \
+            ccache \
+            automake \
+            libtool \
+            qemu-user-binfmt
+          echo "::endgroup::"
+
+      - name: checkout repository
+        uses: actions/checkout@v4
+
+      - name: install cross-toolchain
+        run: |
+          echo "::group::Install cross-toolchain for ${{ matrix.arch }}"
+          sudo apt -y install clang gfortran-14-${{ matrix.target-triple }}
```

Comment:
You still need to install the cross-compiler toolchain (additionally to `clang`).
Also, installing the current compiler version for the respective version of Ubuntu is probably fine.
``` suggestion
          sudo apt -y install clang g++-${{ matrix.target-triple }} gfortran-${{ matrix.target-triple }}
```